src/values/AbstractValue.js

```diff
@@ -516,7 +516,8 @@ export default class AbstractValue extends Value {
     template: PreludeGenerator => ({}) => BabelNodeExpression,
     resultType: typeof Value,
     operands: Array<Value>,
-    optionalArgs?: {| kind?: string, isPure?: boolean, skipInvariant?: boolean |}
+    optionalArgs?: {| kind?: string, isPure?: boolean, skipInvariant?: boolean |},
+    forceHydrateLazyObjects: boolean = false
```

Comment:
I don't fully understand the semantics of `forceHydrateLazyObjects` (I can see how it's all connected, but I don't have a good mental modal for it). Can I consider it a hack in place until we switch to @hermanventer's suggestion?

Reply:
Yes, it is a hack flag telling derive() method to emit that magic comment or not. If we are able to switch to @hermanventer's suggestion we should be able to remove this usage.